File: app/src/main/java/com/example/shop/data/remote/dto/ProductDtos.kt
package com.example.shop.data.remote.dto

import com.example.shop.data.model.Product

data class ProductResponse(
    val id: Int,
    val name: String,
    val price: Double,
    val description: String,
    val imageUrl: String,
    val quantity: Int,
    val categoryId: Int,
    val categoryName: String
)

data class CreateProductRequest(
    val name: String,
    val price: Double,
    val description: String,
    val quantity: Int,
    val categoryId: Int,
    val imageUrl: String? = null
)

data class UpdateProductRequest(
    val name: String,
    val price: Double,
    val description: String,
    val quantity: Int,
    val categoryId: Int,
    val imageUrl: String? = null
)

fun ProductResponse.toProduct(): Product {
    return Product(
        id = id,
        name = name,
        price = price,
        description = description,
        imageUrl = imageUrl,
        quantity = quantity,
        categoryId = categoryId
    )
}

File: backend/ShopApi/Dtos/ProductDtos.cs
using System.ComponentModel.DataAnnotations;

namespace ShopApi.Dtos;

public record ProductResponse(
    int Id,
    string Name,
    decimal Price,
    string Description,
    string ImageUrl,
    int Quantity,
    int CategoryId,
    string CategoryName);

public record CreateProductRequest(
    [Required, StringLength(160, MinimumLength = 2)] string Name,
    decimal Price,
    [Required, StringLength(2000, MinimumLength = 1)] string Description,
    int Quantity,
    int CategoryId,
    [StringLength(500)] string? ImageUrl = null);

public record UpdateProductRequest(
    [Required, StringLength(160, MinimumLength = 2)] string Name,
    decimal Price,
    [Required, StringLength(2000, MinimumLength = 1)] string Description,
    int Quantity,
    int CategoryId,
    [StringLength(500)] string? ImageUrl = null);
